tests/Unit/Evolution/DgSubcell/Test_Projection.cpp

```diff
@@ -0,0 +1,154 @@
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+namespace Tags {
+template <typename Tag>
+struct Prefix : db::PrefixTag {
+  using tag = Tag;
+  using type = typename Tag::type;
+};
+
+struct Scalar : db::SimpleTag {
+  using type = ::Scalar<DataVector>;
+};
+
+template <size_t Dim>
+struct Vector : db::SimpleTag {
+  using type = tnsr::I<DataVector, Dim>;
+};
+}  // namespace Tags
+
+template <size_t MaxPts, size_t Dim, Spectral::Basis BasisType,
+          Spectral::Quadrature QuadratureType>
+void test_project_fd() noexcept {
+  CAPTURE(Dim);
+  CAPTURE(BasisType);
+  CAPTURE(QuadratureType);
+  static_assert(Spectral::maximum_number_of_points<BasisType> >= MaxPts,
+                "Basis does not have enough points. Must have at least 8 "
+                "points per element");
+
+  for (size_t num_pts_1d = std::max(
+           static_cast<size_t>(2),
+           Spectral::minimum_number_of_points<BasisType, QuadratureType>);
+       num_pts_1d < MaxPts + 1; ++num_pts_1d) {
+    CAPTURE(num_pts_1d);
+    const Mesh<Dim> dg_mesh{num_pts_1d, BasisType, QuadratureType};
+    const auto logical_coords = logical_coordinates(dg_mesh);
+    const size_t num_subcells_1d = 2 * num_pts_1d - 1;
+    const Mesh<Dim> subcell_mesh(num_subcells_1d,
+                                 Spectral::Basis::FiniteDifference,
+                                 Spectral::Quadrature::CellCentered);
+    const DataVector nodal_coeffs =
+        TestHelpers::evolution::dg::subcell::cell_values(dg_mesh.extents(0) - 2,
+                                                         logical_coords);
+    // Test projection of a DataVector
+    const DataVector cell_average_times_volume =
+        evolution::dg::subcell::fd::project(nodal_coeffs, dg_mesh,
+                                            subcell_mesh.extents());
+    CHECK_ITERABLE_APPROX(
+        cell_average_times_volume,
+        TestHelpers::evolution::dg::subcell::cell_values(
+            dg_mesh.extents(0) - 2, logical_coordinates(subcell_mesh)));
+
+    // Test projection of a Variables using the return-by-value function
+    Variables<tmpl::list<Tags::Scalar, Tags::Vector<Dim>>> vars(
+        dg_mesh.number_of_grid_points());
+    get(get<Tags::Scalar>(vars)) =
+        TestHelpers::evolution::dg::subcell::cell_values(dg_mesh.extents(0) - 2,
+                                                         logical_coords);
+    for (size_t d = 0; d < Dim; ++d) {
+      get<Tags::Vector<Dim>>(vars).get(d) =
+          (d + 2.0) * TestHelpers::evolution::dg::subcell::cell_values(
+                          dg_mesh.extents(0) - 2, logical_coords);
+    }
+    Variables<tmpl::list<Tags::Scalar, Tags::Vector<Dim>>>
+        cell_average_times_volume_vars = evolution::dg::subcell::fd::project(
+            vars, dg_mesh, subcell_mesh.extents());
+
+    const auto check_impl = [&dg_mesh,
```

Comment:
I suggest `check_impl` -> `check_each_field_in_vars` or `wrap_check_iterable_approx` or something that hints a little more at the contents